### PR TITLE
Check fixed tuple unpacking in super calls

### LINKED ISSUE #553: `super()` calls ignore positional arguments supplied by `*` unpacking
## Summary

The manual `super()` checker ignores positional arguments supplied through `*` unpacking, even when mypy knows the unpacked tuple has exactly one item. The equivalent direct method calls are rejected correctly.

## Reproduction

`pyproject.toml`:

```toml
[tool.mypy]
plugins = ["mypy_strict_kwargs"]
strict = true
```

`example.py`:

```python
class Base:
    def method(self, value: int) -> None:
        pass


class Child(Base):
    def call(self, args: tuple[int]) -> None:
        super().method(*args)
        super().method(*(1,))


args = (1,)
Base().method(*args)
Base().method(*(1,))
Child().call(args=args)
```

Run:

```console
$ mypy --no-incremental example.py
example.py:13: error: Too many positional arguments for "method" of "Base"  [call-arg]
example.py:14: error: Too many positional arguments for "method" of "Base"  [call-arg]
Found 2 errors in 1 file (checked 1 source file)
```

## Expected

All four calls pass one positional argument to `Base.method` and should be rejected consistently. In particular, the calls on lines 8 and 9 should produce the same error as the direct calls on lines 13 and 14.

## Actual

Only the direct calls are rejected. `_call_disallows_positional_argument` considers only `ArgKind.ARG_POS` actual arguments and skips `ArgKind.ARG_STAR`, so both `super()` calls pass silently.

Without the plugin, mypy succeeds; the example runs successfully.

## Environment

- mypy-strict-kwargs commit: `14a7b62` (`origin/main` on 2026-07-16)
- mypy: 2.3.0
- Python: 3.11.14

## Suggested fix direction

Make the `super()` path use the same typed argument mapping as ordinary method calls, including fixed-length tuple unpacking. The class-definition hook runs before type checking, so handling this reliably may require moving the check to a hook that has access to inferred argument types rather than treating every `ARG_STAR` as unknown.

### PR DESCRIPTION
**Pull request by adamtheturtle**
Closes #553.

## What changed

- Track fixed tuple lengths from function parameter annotations while traversing class bodies.
- Expand fixed tuple literals and fixed-tuple parameters in the manual super argument mapper.
- Keep unknown-length tuple unpacking conservative and preserve variadic positional formals.
- Add regression coverage and a changelog fragment.

## Why

The public typed call hooks are not invoked for super method calls, so the existing pre-type-check checker must retain the fixed arity information available in syntax and annotations.

## Validation

- uv run --extra=dev pytest -q (45 passed)
- repository pre-commit and pre-push checks passed

---

> [!NOTE]
> **Low Risk**
> Scoped to the plugin’s AST walk and super-call validation; behavior changes only affect error reporting for unpacked super arguments, with new tests covering edge cases.
> 
> **Overview**
> Extends the **mypy_strict_kwargs** manual `super().method(...)` checker so `*args` is treated as multiple positional arguments when arity is known from syntax or annotations.
> 
> Call collection now uses **`_CollectedCalls`** to record, per call site, fixed-length tuple parameters (from `tuple[...]` / `Tuple[...]` annotations) and to respect inner-scope name shadowing. **`_call_disallows_positional_argument`** expands `ARG_STAR` via tuple literals, annotated fixed tuples, or stays conservative (0 extra positions) for lists and unknown unpacks; variadic `*values` formals still absorb expansions without false positives.
> 
> Regression YAML cases cover `super().method(*args)`, nested functions, and shadowed parameter names; changelog fragment added for #553.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit d0d05f8628960b464ac201135c7a379ca7337dfd. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/mypy_strict_kwargs/plugin.py [hunk 1]
```diff
@@ -461,19 +492,50 @@ def _collect_call_exprs_from_statement(  # noqa: C901, PLR0912, PLR0915  # pylin
...
+                annotation=argument.type_annotation,
+            )
+        )
+        is not None
+    }
+    for call in calls[first_body_call_index:]:
+        calls.fixed_tuple_lengths.setdefault(
+            call,
+            fixed_tuple_lengths,
+        )
```
[the remark this answers is not in the document]
Comment (adamtheturtle) [reply src/mypy_strict_kwargs/plugin.py]: Fixed in e5b3d4a: enclosing-scope fixed-tuple metadata now propagates into nested functions/lambdas, while inner-scope parameters still shadow enclosing parameters of the same name (covered by `super_method_fixed_tuple_unpacking_nested_scope` and `super_method_fixed_tuple_unpacking_inner_scope_shadows`).
